packages/serve/test/auth/cannerPATAuthenticator.spec.ts

```diff
@@ -0,0 +1,277 @@
+  // Arrange
+  const ctx = {
+    ...sinon.stubInterface<KoaContext>(),
+    request: {
+      ...sinon.stubInterface<Request>(),
+      headers: {
+        ...sinon.stubInterface<IncomingHttpHeaders>(),
+        authorization: `Canner-PAT eyValidToken`,
+      },
+    },
+  } as KoaContext;
+  // mock axios.post method get non 200 response from remote server then axios throw error
+  mockedAxios.post.mockRejectedValue({
+    request: 'mock request',
+    response: {
+      data: { error: 'mock response error' },
+      status: 500,
+      statusText: 'string',
+      headers: {},
+      config: {},
+    } as any,
+    isAxiosError: true,
+    toJSON: () => ({}),
+    message: 'An error occurred!',
+    name: 'Error',
+  });
+  const authenticator = await getStubAuthenticator(mockOptions);
+
+  // Act
+  const res = await authenticator.authCredential(ctx);
+
+  // Assert
+  await expect(res).toEqual(
+    expectFailed(
+      'Failed to fetch user info from canner server: response status: 500, response data: {"error":"mock response error"}'
+    )
+  );
+}, 10000);
+
+it('Test to auth credential failed when request header not exist "authorization" key', async () => {
+  // Arrange
+  const ctx = {
+    ...sinon.stubInterface<KoaContext>(),
+    request: {
+      ...sinon.stubInterface<Request>(),
+      headers: {
+        ...sinon.stubInterface<IncomingHttpHeaders>(),
+      },
+    },
+  } as KoaContext;
+  const authenticator = await getStubAuthenticator(mockOptions);
+
+  // Act
+  const result = await authenticator.authCredential(ctx);
+
+  // Assert
+  expect(result).toEqual(expectIncorrect);
+});
+
+it('Should auth credential failed when request header "authorization" not start with "canner-pat"', async () => {
+  // Arrange
+  const ctx = {
+    ...sinon.stubInterface<KoaContext>(),
+    request: {
+      ...sinon.stubInterface<Request>(),
+      headers: {
+        ...sinon.stubInterface<IncomingHttpHeaders>(),
+        authorization: 'Incorrect-Prefix 1234567890',
+      },
+    },
+  } as KoaContext;
+  const authenticator = await getStubAuthenticator(mockOptions);
+
+  // Act
+  const result = await authenticator.authCredential(ctx);
+
+  // Assert
+  expect(result).toEqual(expectIncorrect);
+});
+
+it('Should auth credential failed when the PAT token in the authorization header does not pass the canner host', async () => {
+  // Arrange
+  const ctx = {
+    ...sinon.stubInterface<KoaContext>(),
+    request: {
+      ...sinon.stubInterface<Request>(),
+      headers: {
+        ...sinon.stubInterface<IncomingHttpHeaders>(),
+        authorization: `Canner-PAT ${invalidToken}`,
+      },
+    },
+    set: sinon.stubInterface<BaseResponse>().set,
+  };
+  const mockResolveValue = {
+    status: 401,
+    data: { error: 'invalid token from canner' },
+  };
+  const authenticator = await getStubAuthenticator(
+    mockOptions,
+    mockResolveValue
+  );
```

Comment:
Just curious ? Why do  you pass the `mockResolveValue` be a resolved value for stub `fetchCannerUser` ? 

Seems the `fetchCannerUser` won't return the data whose status is `401`, because when Axios get the `401,` or I miss some situation from canner enterprise ?

Reply:
I'll remove the test cases that are testing the remote response whose status is not 200, thanks